pkg/crypto/x509/x509.go

```diff
@@ -385,6 +387,41 @@ func NewCertificateAndKeyFromFiles(crt, key string) (p *PEMEncodedCertificateAnd
 	return p, nil
 }
 
+// NewCSRAndIdentity generates and PEM encoded certificate and key, along with a
+// CSR for the generated key.
+func NewCSRAndIdentity(hostname string, ips []net.IP) (csr *CertificateSigningRequest, identity *PEMEncodedCertificateAndKey, err error) {
+	var key *Key
+	key, err = NewKey()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	identity = &PEMEncodedCertificateAndKey{
+		Key: key.KeyPEM,
+	}
+
+	pemBlock, _ := pem.Decode(key.KeyPEM)
+	if pemBlock == nil {
+		return nil, nil, fmt.Errorf("failed to decode key")
+	}
+
+	keyEC, err := x509.ParseECPrivateKey(pemBlock.Bytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	opts := []Option{}
+	opts = append(opts, DNSNames([]string{hostname}))
```

Comment:
I think it's fine for this pass, but I'm expecting this will need to be more flexible in the future.

Reply:
Yep exactly. This was part of the existing setup.